src/webgpu/util/texture/image_copy.ts

```diff
@@ -25,7 +25,7 @@ export function dataBytesForCopy(
   format: SizedTextureFormat,
   copyExtentValue: GPUExtent3D,
   { method }: { method: ImageCopyType }
-): { minDataSize: number; valid: boolean } {
+): { validMinDataSize: number | undefined; bestGuessMinDataSize: number } {
```

Comment:
"Guess" is unfortunate here, because I'm sure some tests will rely on the value of the guess. I actually liked the previous return value that said 1) whether your copy was valid 2) what the minimum size of the buffer should be. Also we can see in this commit that no test uses both `validMinDataSize` and `bestGuessMinDataSize`.

Reply:
The reason for guessing is that a number of validation tests are attempting to test conditions on bytesPerRow/rowsPerImage but we want to make sure we don't *also* trigger an error on "buffer wasn't big enough".

This is really just a refactor; same semantics but exposed differently. Exposing validMinDataSize and bestGuessMinDataSize separately ensures callers don't forget to `assert(valid)` if they should.

I'm not sure what would be better than this. Two separate entry points, one which returns best guess and one which throws if not valid?

Reply:
Made two helper entry points.